### PR TITLE
s32k1xx: added way to get the reset cause in a proc fs

### PR DESCRIPTION
**Pull request by cisvanmierlo**
## Summary
Added an optional proc fs to get the reset cause, which can be used in the application to act on. 

## Impact
minimal, only added the optional proc fs to current system.

## Testing
Tested on RDDRONE-BMS772, it works.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@gustavonihei @Ouss4 since the reset cause is very important information, should we standardize all possible cause and require SoC convert the private definition to the standard one? Another question is what's the best method to expose this crucial info to userspace monitor tool.

**Comment by Ouss4** [comment]
> since the reset cause is very important information, should we standardize all possible cause and require SoC convert the private definition to the standard one?

We can have a subset of the rest causes that are common, but we'll still have a few that are specific to a particular SoC.  How would we handle that?  I guess we can do something as easy as considering the values above the common set max to be chip specific.

As for the best way to get this info, I personally don't know which one is best between procfs and boardctl, they actually both tend to go beyond their scope (see https://github.com/apache/incubator-nuttx/issues/1132 for instance.)
We already use broadctl for stuff like uniqueid and actually reseting the board, the reset cause seems to be similar to these, is there any reason to use procfs here?

**Comment by xiaoxiang781216** [comment]
> > since the reset cause is very important information, should we standardize all possible cause and require SoC convert the private definition to the standard one?
> 
> We can have a subset of the rest causes that are common, but we'll still have a few that are specific to a particular SoC. How would we handle that? I guess we can do something as easy as considering the values above the common set max to be chip specific.
> 

Yes, we can define the common cause, and reserve a range for SoC specific cause, just like IOCTL.

> As for the best way to get this info, I personally don't know which one is best between procfs and boardctl, they actually both tend to go beyond their scope (see #1132 for instance.)
> We already use broadctl for stuff like uniqueid and actually reseting the board, the reset cause seems to be similar to these, is there any reason to use procfs here?

procfs is good for the interactive usage, but hard to parse/use by the tool and also increase the noticeable size for tiny device.

**Comment by cisvanmierlo** [comment]
Hi, this is for the S32K1xx only and this is needed for the BMS application. We could always adjust it when there is a generic interface for the reset cause.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@Ouss4 let's merge it first?

**Review by Ouss4 — APPROVED** [review]
